### PR TITLE
Bugfixes (Game version 0.10.29.22015) 

### PR DESCRIPTION
**Pull request by starfi5h**
Fix some bugs found in Nebula v0.9.2

- Lock packetProcessor.Write to prevent thread error
- Fix NRE in `EnemyDFGroundSystem.KeyTickLogic (System.Int64 time);(IL_0929)`
- Fix NRE in `DFSTurretComponent.InternalUpdate (PrefabDesc pdesc);(IL_0017)`
- Fix IdxErr in `EnemyFormation.RemoveUnit (System.Int32 port) [0x0000a] ;IL_000A`
- Fix IdxErr in `UIZS_FighterEntry._OnUpdate () [0x001bb] ;IL_01BB`
- Fix `UIStorageGrid` error in player inventory of client
- Fix mecha hp and power modifier don't set correctly when importing on client
- Try to fix ILS related issues

Headless mode fix:
- Fix errors in `SectorModel.LateUpdate`
- Try to fix the issue of planet shield shape GPU calculation by assuming the shield shape is sphere.

### CODE REVIEW

File: NebulaPatcher/Patches/Dynamic/SpaceSector_Patch.cs [hunk 1]
```diff
@@ -65,4 +66,26 @@ public static bool TryCreateNewHive_Prefix(StarData star)
...
+
+        // Fix NRE in DFSTurretComponent.InternalUpdate (PrefabDesc pdesc);(IL_0017)
+        for (var enemyId = 1; enemyId < __instance.enemyCursor; enemyId++)
+        {
+            ref var enemy = ref __instance.enemyPool[enemyId];
+            if (enemy.id != enemyId) continue;
+
+            if (SpaceSector.PrefabDescByModelIndex[enemy.modelIndex] == null)
+            {
+                var msg = $"Remove SpeaceSector enemy[{enemyId}]: modelIndex{enemy.modelIndex}";
```
Comment (PhantomGamers) [inline NebulaPatcher/Patches/Dynamic/SpaceSector_Patch.cs:84]:
```suggestion
                var msg = $"Remove SpaceSector enemy[{enemyId}]: modelIndex{enemy.modelIndex}";
```

File: NebulaNetwork/PacketProcessors/Logistics/ILSIdleShipBackToWorkProcessor.cs [hunk 2]
```diff
@@ -16,18 +17,18 @@ public class ILSIdleShipBackToWorkProcessor : PacketProcessor<ILSIdleShipBackToW
...
             return;
         }
 
-        if (!IsClient)
+        try
         {
-            return;
+            ILSShipManager.IdleShipGetToWork(packet);
         }
-        using (Multiplayer.Session.Factories.IsIncomingRequest.On())
```
Comment (PhantomGamers) [inline NebulaNetwork/PacketProcessors/Logistics/ILSIdleShipBackToWorkProcessor.cs:28]: Is removing this using also intentional?

**Review by PhantomGamers — APPROVED** [review]
If the removal mentioned in the other comment is intentional this looks good to me

File: NebulaNetwork/PacketProcessors/Logistics/ILSIdleShipBackToWorkProcessor.cs (hunk 2 above)
Comment (starfi5h) [reply NebulaNetwork/PacketProcessors/Logistics/ILSIdleShipBackToWorkProcessor.cs:28]: yes, it's intentional.
There isn't methods on the call path that require the locking of `Factories.IsIncomingRequest`

**Pull request merged**